### DISCUSSION TITLE
(solved) Fix database won't start after update (The database import failed.)

### DISCUSSION THREAD

**Discussion by WuSiYu** [nextcloud/all-in-one#2507] [Questions] [answered]
### Steps to reproduce
1. Stop and Start all containers for update (last time few month ago)

### Expected behavior 
All containers enter "Running" state.

### Actual behavior 
Database container stuck on Starting state for a long time, log shows 
```
The database import failed. Please restore a backup and try again.
For further clues on what went wrong, look at the logs above.
```

In the logs above, I found:
```
initdb: error: directory "/var/lib/postgresql/data" exists but is not empty
initdb: detail: It contains a dot-prefixed/invisible file, perhaps due to it being a mount point.
```

Then I found and only found a `.postgresql.conf.un~` file in `nextcloud_aio_database` volume, which is `/var/lib/postgresql/data` in the database container.

I delete this file (and remove `database-import.log` and `import.failed` file in `nextcloud_aio_database_dump` volume if necessary), and restart the data container, then it start correctly and all the thing going right.

I record this for others may have the same issue and it may be helpful.

### Host OS 
Unraid 6.12.0-rc5

#### Nextcloud AIO version 
Nextcloud AIO v5.0.0

#### Current channel 
latest (Nextcloud 25.0.6)

**Comment by WuSiYu** [comment]
(I am not very sure it's whether a bug of nextcloud-aio itself, feel free to close this if you think it's not.)

**Comment by szaimen** [comment]
I don't think this is a bug of nextcloud-aio so moving to discussions

**Comment by szaimen** [comment, marked as answer]
Solution:

> Then I found and only found a `.postgresql.conf.un~` file in `nextcloud_aio_database` volume, which is `/var/lib/postgresql/data` in the database container.

> I delete this file (and remove `database-import.log` and `import.failed` file in `nextcloud_aio_database_dump` volume if necessary), and restart the data container, then it start correctly and all the thing going right.

> I record this for others may have the same issue and it may be helpful.

**Comment by guhlke** [comment]
Many thanks for sharing, you saved my life.
Stopping all containers, deleting files _database-import.log_ and _import.failed_ in the /var/lib/docker/volumes/nextcloud_aio_database_dump/_data folder
Starting containers solved the issue for me as well.

**Comment by Felix2M** [comment]
After the last update I also had the error: `The database import failed. Please restore a backup and try again.`
But I had no special error message like:

`initdb: error: directory "/var/lib/postgresql/data" exists but is not empty`
`initdb: detail: It contains a dot-prefixed/invisible file, perhaps due to it being a mount point.`

I followed the instructions to delete `import.failed`, `database-import.log` and `update.failed` (appdata folder) and now it seems like my database is gone. Also my backup did not work and I am only left with the backup of april and the following occ commands ar edoing nothing:
```
occ maintenance:repair
occ files:scan-app-data
occ files:scan --all
```

I only get the error message:
```
Next OC\DB\Exceptions\DbalException: An exception occurred while executing a query: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "oc_appconfig" does not exist
LINE 1: ... "appid", "configkey", "configvalue", "type" FROM "oc_appcon...
                                                             ^ in /var/www/html/lib/private/DB/Exceptions/DbalException.php:56
```

and

`relation "oc_appconfig" does not exist at character 57`

![grafik](https://github.com/user-attachments/assets/3259e7fc-866b-4136-ba25-d5d8c8ad30ce)
